src/pages/tutorials/index.tsx

```diff
@@ -42,88 +45,101 @@ const DEFAULT_HARDNESS = {
 }
 
 function TutorialItem({ item }: { item: Project }) {
-  if (!item.tutorial) return
-  const { githubId } = item
+  const navigate = useNavigate()
+  const setTutorialToc = useSetAtom(tutorialTocAtom)
+
+  const handleNavigation = async () => {
+    const data = await getTutorialToc(item.owner.login, item.name)
+    setTutorialToc(data)
+    // const computedPath = `/tutorial/${item.githubId}/${item.owner.login}/${item.name}/${encodeURIComponent(
+    //   data[0].path,
+    // )}`
+    const path = data[0].path?.match(/^(.*?).md$/)?.[1]
+    const computedPath = `/tutorial/${item.githubId}/rust-lang/async-book/${encodeURIComponent(path || '')}`
+    navigate(computedPath)
+  }
+
   return (
```

Comment:
why revmoe <Link>, I think it import many unrelevant lines  of code?

Reply:
Please also handle the logic to make sure the default target url is the first chapter of the summary.md.

Reply:
> Please also handle the logic to make sure the default target url is the first chapter of the summary.md.

This calculation relies on an I/O request, and performing it during the render phase would trigger a request for each document to fetch the necessary data. I believe this would introduce unnecessary overhead with multiple computations and requests. A better approach would be to handle this calculation asynchronously during the user's click event, making the request only when needed.
Alternatively, the backend should provide an additional field that returns the corresponding target URL and data, so I don’t have to make another request to fetch it.

Reply:
I have second thought. Seperate the list and content is very nature.
In current stage, we don't have time to merge all necessary apis to improve only a little performance.